docs/integration-tests.md

```diff
@@ -0,0 +1,72 @@
+
+### General Cypress Downsides
+- Cypress tests are e2e tests which generally means they are slow to run and will be more flakey than unit level tests. This is generally true of the nature of all e2e tests.
+- We are still coupled to a system for running our tests, now our tests will be coupled to Cypress and the Web Browser.
+- When a test fails in cypress it will not be clear what piece of functionality is broken, what will fail is a DOM element that could not be found. In an ideal world when a test fails the developer can know exactly why it is failing.
+
+
+## A Path Forward
+
+Overall we feel cypress is a better testing approach compared to our home brewed integration test suite.  Cypress is also very common in the industry with 45k github stars, it's still activelty maintained, and it's backed by an actual company.  Because of this, onboarding new developers to the Dawson project is easier since cypress is well known.  The ability to watch cypress videos when tests fail is invaluable in debugging flaky tests, and writing these tests is a lot easier than our current approach. The downside of coupling ourselves to a 3rd party framework for testing key functionality is not as great as the downside of needing to create/maintain our own custom version of testing for the application.
+
+We propose that all future integration and e2e tests are written in cypress.  We feel this will allow us to not only slowly decouple from cerebral, but also build up a more realiable test suite.  We also recommend that any existing integration tests that are flaky should be rewritten in cypress.  We've had a lot of flaky integration tests fail on github actions and circle, and refactoring them to cypress might be a good solution to make our test suites more stable.
+
+At some point we may decide to slowly refactor existing integration tests into cypress tests with the hopes to be able to remove cerebral from our system and not lose all our test coverage.
+
+## Best Practices
+
+In order to write a realiable cypress test suites, there are some best practices we should follow that our outlined in the cypress documentation and also some best practices we have learned from trying to write realiable tests.
+
+- every `it` statement should be a repeatable test that should be able to run in isolation.  This means your `it` statement shouldn't depend on a previous it statement to pass.  If you write tests that depend on previous it statements, your tests can become flaky.  This also goes against the best practices outlined in the cypress docs.
+- dom elements should be accessed using our custom cypress command `cy.getByTestId("")`.  This means any element you need to click or find in the UI must have a `data-testid` attribute.  The reason test ids are prefered is that some attributes such as classes and ids can accidently be changed when making style changes; whereas, it's obvious when you see a data-testid that a cypress test uses that element.
```

Comment:
Looks like you've gotten rid of the `cy.getTestById("")` in favor of getting the item directly via the `[data-testid]`